src/ralph/sim_cards/models.py

```diff
@@ -0,0 +1,94 @@
+from django.conf import settings
+from django.db import models
+from django.core.validators import RegexValidator
+from django.utils.translation import ugettext_lazy as _
+
+
+from ralph.back_office.models import Warehouse
+from ralph.lib.mixins.models import AdminAbsoluteUrlMixin, NamedMixin
+
+
+class CellularCarrier(AdminAbsoluteUrlMixin, NamedMixin, models.Model):
+    pass
+
+
+class SIMCard(AdminAbsoluteUrlMixin, models.Model):
+    pin1 = models.CharField(
+        max_length=8, null=True, blank=True, help_text=_('required numeric '
+                                                         'characters only '
+                                                         '(max length 16)'),
+        validators=[
+            RegexValidator(regex='^\d{1,8}$',
+                           message='required numeric '
+                                   'characters only (max length 8)'
+                           )
+        ]
+    )
+    puc1 = models.CharField(
+        max_length=16, help_text=_('required numeric '
+                                   'characters only '
+                                   '(max length 16)'),
+        validators=[
+            RegexValidator(regex='^\d{1,8}$',
+                           message='required numeric '
+                                   'characters only (max length 16)'
+                           )
+        ])
+    pin2 = models.CharField(
+        max_length=8, null=True, blank=True, help_text=_('asd'),
+        validators=[
+            RegexValidator(regex='^\d{1,8}$',
+                           message='required numeric '
+                                   'characters only (max length 8)'
+                           )
+        ]
+    )
+    puc2 = models.CharField(
+        max_length=16,
+        validators=[
+            RegexValidator(regex='^\d{1,8}$',
+                           message='required numeric '
```

Comment:
Better to wrap all user-facing strings with _()

Reply:
done